core/state_processor.go

```diff
@@ -73,6 +73,30 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	blockContext := NewEVMBlockContext(header, p.bc, nil)
 	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, p.config, cfg)
 	// Iterate over and process the individual transactions
+
+	if p.config.Optimism != nil && p.config.Optimism.Enabled {
```

Comment:
Is there a way to merge these two loops and only apply the L1Cost option if it's provided?
This code is pretty stables, but I don't love having the duplication here, particularly when upstream goes for an update.